### ISSUE TITLE
Running bin/logstash with an unsupported command yield `too many arguments`

### ISSUE THREAD

**Issue by ph** [elastic/logstash#5748]
This was originaly found by @LeeDr on windows but I can reproduce it on OS X.

If you start the `bin/logstash` executable with an invalid command it will display `ERROR: too many arguments`, it should instead display something like `version is an unknown command"


**Steps to reproduces:**
```
~/e/logstash git:feature/docgen-gem ❯❯❯ bin/logstash invalidcommand                                                                                                                                                                                                                                                          
ERROR: too many arguments

See: 'bin/logstash --help'
~/e/logstash git:feature/docgen-gem ❯❯❯

**Comment by jsvd** [comment]
Interesting, this is an exposure of clamp's internal behaviour, it sequentially consumes known flags and then checks if there's anything left which is unknown, giving this error. looking for a way to catch that..

**Comment by jsvd** [comment]
closed by #5772
